multiwallet.py

```diff
@@ -487,6 +516,55 @@ def do_generate_seed(self, arg):
         print_yellow("Copy-paste this into Specter-Desktop:")
         print_green(key_record)
 
+    def do_generate_seed_from_dice(self, arg):
+        """Calculate bitcoin public and private key information from BIP39 words created from dice rolls"""
+
+        blue_fg("Generates a mnemonic from 6-sided dice rolls.")
+        num_words = _get_num_words()
+        dice_vals = _get_dice_values()
+
+        if self.ADVANCED_MODE:
+            password = _get_password()
+        else:
+            password = ""
+
+        if _get_bool(prompt="Use Mainnet?", default=False):
+            network = "mainnet"
+        else:
+            network = "testnet"
+
+        if self.ADVANCED_MODE:
+            path_to_use = _get_path(network=network)
+            use_slip132_version_byte = _get_bool(
+                prompt="Encode with SLIP132 version byte?", default=True
+            )
+        else:
+            path_to_use = None  # buidl will use default path
+            use_slip132_version_byte = True
+
+        words = dice_rolls_to_mnemonic(dice_vals, num_words)
+        hd_priv = HDPrivateKey.from_mnemonic(
+            mnemonic=words,
+            password=password.encode(),
+            network=network,
+        )
+
+        key_record = hd_priv.generate_p2wsh_key_record(
+            bip32_path=path_to_use, use_slip132_version_byte=use_slip132_version_byte
+        )
+
+        print(yellow_fg("SECRET INFO") + red_fg(" (guard this VERY carefully)"))
+        print_green(
+            f"Dice rolls used: {dice_vals}"
+            f"\nFull ({len(words.split())} word) mnemonic (including last word): {words}"
+        )
+        if password:
+            print_green(f"Passphrase: {password}")
+
+        print_yellow(f"\nPUBLIC KEY INFO ({network})")
+        print_yellow("Copy-paste this into Specter-Desktop:")
+        print_green(key_record)
+
```

Comment:
This has a lot of overlap with `do_generate_seed()`, what do you think about refactoring the code so that the shared code only lives in one place? For example, I noticed that yours doesn't include the footgun warning that is in `do_generate_seed`. DRYing it out would seem to make more sense.